tests/utils/stl/test/file_parsing.py

```diff
@@ -0,0 +1,153 @@
+from collections import namedtuple
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Dict, FrozenSet, List, Optional, Set, Tuple, Union
+import itertools
+import os
+import re
+
+import lit.Test
+
+_envlst_cache = dict()
+_preprocessed_file_cache = dict()
+_expected_result_entry_cache = dict()
+
+
+@dataclass
+class _TmpEnvEntry:
+    tags: Set[str] = field(default_factory=set)
+    env: Dict[str, str] = field(default_factory=dict)
+
+
+@dataclass(frozen=True)
+class EnvEntry:
+    def __init__(self, tmp_env: _TmpEnvEntry):
+        object.__setattr__(self, "_tags", frozenset(tmp_env.tags))
+        object.__setattr__(self, "_env_keys", tuple(tmp_env.env.keys()))
+        object.__setattr__(self, "_env_vals", tuple(tmp_env.env.values()))
+
+    def getEnvVal(self, key: str, default: Optional[str] = None)\

```

Comment:
not a huge fan of the \ line escape, but if autopep8 or something did it then fine

Reply:
falke8 yelled at me for it and I chose to fix it that way. If you'd like to suggest a different way of breaking up the line I'd probably take it.

Reply:
I'd like to suggest ` \` =)
```suggestion
    def getEnvVal(self, key: str, default: Optional[str] = None) \
```

Reply:
(I'm going to stop commenting occurrences now and assume you can search for `\S\\$`.)